reV/supply_curve/sc_aggregation.py

```diff
@@ -1093,45 +1112,56 @@ def run_serial(
-                            resolution=resolution,
-                            exclusion_shape=exclusion_shape,
-                            power_density=fh.power_density,
-                            args=args,
-                            excl_dict=excl_dict,
-                            inclusion_mask=gid_inclusions,
-                            excl_area=excl_area,
-                            close=False,
-                            friction_layer=fh.friction_layer,
-                            cap_cost_scale=cap_cost_scale,
-                            recalc_lcoe=recalc_lcoe,
-                        )
+                zones = cls._get_gid_zones(
+                    excl_fpath, zones_dset, gid, slice_lookup
+                )
+                zone_ids = np.unique(zones[zones != 0]).tolist()
 
-                    except EmptySupplyCurvePointError:
-                        logger.debug("SC point {} is empty".format(gid))
-                    else:
-                        pointsum['res_class'] = ri
+                for ri, res_bin in enumerate(res_class_bins):
+                    for zone_id in zone_ids:
+                        zone_mask = zones == zone_id
+                        try:
+                            pointsum = GenerationSupplyCurvePoint.summarize(
+                                gid,
+                                fh.exclusions,
+                                fh.gen,
+                                tm_dset,
+                                gen_index,
+                                res_class_dset=res_data,
+                                res_class_bin=res_bin,
+                                cf_dset=cf_dset,
+                                lcoe_dset=lcoe_data,
+                                h5_dsets=h5_dsets_data,
+                                data_layers=fh.data_layers,
+                                resolution=resolution,
+                                exclusion_shape=exclusion_shape,
+                                power_density=fh.power_density,
+                                args=args,
+                                excl_dict=excl_dict,
+                                inclusion_mask=gid_inclusions,
+                                excl_area=excl_area,
+                                close=False,
+                                friction_layer=fh.friction_layer,
+                                cap_cost_scale=cap_cost_scale,
+                                recalc_lcoe=recalc_lcoe,
+                                zone_mask=zone_mask,
+                            )
 
-                        summary.append(pointsum)
-                        logger.debug(
-                            "Serial aggregation completed gid {}: "
-                            "{} out of {} points complete".format(
-                                gid, n_finished, len(gids)
+                        except EmptySupplyCurvePointError:
+                            logger.debug("SC point {} is empty".format(gid))
```

Comment:
Leaving this here now so I don't forget later. Let's add the zone ID to this logger message, otherwise the logs look something like this: 
![image](https://github.com/user-attachments/assets/9c0e380d-3ed3-4490-b014-7d1886b51ed6)

Reply:
Added this since I was already working in the code